storage/atsd/atsd.go

```diff
@@ -0,0 +1,400 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package atsd
+
+import (
+	"flag"
+	"fmt"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	info "github.com/google/cadvisor/info/v1"
+	"github.com/google/cadvisor/manager"
+	"github.com/google/cadvisor/storage"
+
+	"github.com/BurntSushi/toml"
+	atsdHttp "github.com/axibase/atsd-api-go/http"
+	atsdStorageDriver "github.com/axibase/atsd-storage-driver/storage"
+	"os"
+	"strconv"
+)
+
+const (
+	startDelay           = 15 * time.Second       //waiting to store enough data for all entities before send
+	timestampPeriodError = 250 * time.Millisecond //time error accumulated during housekeeping
+
+	metricPrefix = "cadvisor"
+
+	// metric groups for a deduplication
+	cpuGroup       = "cpu"
+	ioGroup        = "io"
+	memoryGroup    = "memory"
+	taskGroup      = "task"
+	networkGroup   = "network"
+	filesytemGroup = "filesystem"
+)
+
+var (
+	urlString       = flag.String("storage_driver_atsd_url", "", "ATSD server http/https endpoint")
+	writeHost       = flag.String("storage_driver_atsd_write_host", "", "ATSD server TCP/UDP destination, formatted as host:port")
+	protocol        = flag.String("storage_driver_atsd_write_protocol", "", "transfer protocol. Possible settings: http/https, udp, tcp")
+	connectionLimit = flag.Uint("storage_driver_atsd_connection_limit", 1, "ATSD storage driver TCP connection count")
```

Comment:
It's a debateable topic, but there are some pretty good arguments here: https://www.reddit.com/r/golang/comments/2q5vdu/int_vs_uint/

In your case the value must be > 0, so you already need to do some validation (since uint can be 0), so you might as well use an int.